awswrangler/_data_types.py

```diff
@@ -556,7 +556,7 @@ def pyarrow2pandas_defaults(use_threads: Union[bool, int], kwargs: Optional[Dict
         "use_threads": use_threads,
         "split_blocks": True,
         "self_destruct": True,
-        "ignore_metadata": True,
+        "ignore_metadata": False,
```

Comment:
This would reinstate the index recovery behaviour we had in 2.x. The question is whether this would be considered a bug fix or a breaking change for 3.0.0 users